sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -279,6 +280,49 @@ case class Ceil(child: Expression) extends UnaryMathExpression(math.ceil, "CEIL"
   override protected def withNewChildInternal(newChild: Expression): Ceil = copy(child = newChild)
 }
 
+object Ceil {
+  def apply(child: Expression, scale: Expression): Expression = {
+    val scaleV = scale.eval(EmptyRow)
+    if (scaleV == null) throw new AnalysisException("Scale parameter can not be null")
+
+    scaleV.asInstanceOf[Int] match {
+      case 0 => Ceil(child)
+      case x if x < 0 => RoundCeil(Cast(child, LongType), scale)
+      case _ => RoundCeil(child, scale)
+    }
+  }
+}
+
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Returns the smallest integer not smaller than `expr`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(-0.1);
+       0
+      > SELECT _FUNC_(5);
+       5
+      > SELECT _FUNC_(3.1411, 3);
```

Comment:
can we also show an example with negative scale?

Reply:
yes